File: java/11-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length - 1;

        int max = 0;
        while (left < right) {
            int size = (right - left) * Math.min(height[left], height[right]);
            max = Math.max(max, size);
            if (height[left] > height[right]) {
                right--;
            } else {
                left++;
            }
        }

        return max;
    }
}

File: c++/11-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {

        int volume = 0;

        int left = 0;
        int right = height.size()-1;
        int tmp = 0;
        int gap = right;

        while (gap > 0) {
            
            if (height[left] < height[right]) {
                tmp = gap*height[left];
                left++;
            } else {
                tmp = gap*height[right];
                right--;
            }
            volume = tmp > volume ? tmp : volume;
            gap--;
        }
        return volume;
        
    }
};